src/core_plugins/metrics/public/components/custom_color_picker.js

```diff
@@ -0,0 +1,199 @@
+      'disableAlpha': {
+        color: {
+          width: '22px',
+        },
+        alpha: {
+          display: 'none',
+        },
+        hue: {
+          marginBottom: '0px',
+        },
+        swatch: {
+          width: '10px',
+          height: '10px',
+          marginTop: '0px',
+        },
+      },
+    }, this.props);
+
+    const handleSwatchChange = (data) => {
+      if (data.hex) {
+        color.isValidHex(data.hex) && this.props.onChange({
+          hex: data.hex,
+          source: 'hex',
+        });
+      } else {
+        this.props.onChange(data);
+      }
+    };
+
+    const swatches = this.props.colors.map((c) => {
+      return (
+        <CompactColor
+          key={c}
+          color={c}
+          onClick={handleSwatchChange}/>
+      );
+    });
+
+    return (
+      <div style={styles.picker} className="custom-picker">
+        <div style={styles.saturation}>
+          <Saturation
+            style={styles.Saturation}
+            { ...this.props }
+            pointer={ChromePointerCircle}
+            onChange={this.handleChange}
+          />
+        </div>
+        <div style={styles.body}>
+          <div style={styles.controls} className="flexbox-fix">
+            <div style={styles.color}>
+              <div style={styles.swatch}>
+                <div style={styles.active} />
+                <Checkboard />
+              </div>
+            </div>
+            <div style={styles.toggles}>
+              <div style={styles.hue}>
+                <Hue
+                  style={styles.Hue}
+                  {...this.props}
+                  pointer={ChromePointer}
+                  onChange={this.handleChange}
+                />
+              </div>
+              <div style={styles.alpha}>
+                <Alpha
+                  style={styles.Alpha}
+                  {...this.props}
+                  pointer={ChromePointer}
+                  onChange={this.handleChange}
+                />
+              </div>
+            </div>
+          </div>
+          <ChromeFields
+            {...this.props}
+            onChange={this.handleChange}
+            disableAlpha={this.props.disableAlpha}
+          />
+          <div style={styles.swatches} className="flexbox-fix">
+            {swatches}
+          </div>
+        </div>
+      </div>
+    );
+  }
+}
+
+CustomColorPicker.defaultProps = {
+  colors: ['#4D4D4D', '#999999', '#FFFFFF', '#F44E3B', '#FE9200', '#FCDC00',
+    '#DBDF00', '#A4DD00', '#68CCCA', '#73D8FF', '#AEA1FF', '#FDA1FF',
+    '#333333', '#808080', '#cccccc', '#D33115', '#E27300', '#FCC400',
+    '#B0BC00', '#68BC00', '#16A5A5', '#009CE0', '#7B64FF', '#FA28FF',
+    '#0F1419', '#666666', '#B3B3B3', '#9F0500', '#C45100', '#FB9E00',
+    '#808900', '#194D33', '#0C797D', '#0062B1', '#653294', '#AB149E',
+  ],
+};
+
+CustomColorPicker.propTypes = {
+  color            : PropTypes.string,
+  onChangeComplete : PropTypes.func,
+  onChange         : PropTypes.func
```

Comment:
Not a fan of this aligning. This is something that easily breaks unless eslint or something like that automatically fixes it on every commit (e.g. let's say you search and replace a variable name etc)